status/status.go

```diff
@@ -0,0 +1,91 @@
+package status
+
+import "sync"
+
+// Status represents a status update from an action within Empire.
+type Status interface {
+	// Returns a human readable string about the status update.
+	String() string
+}
+
+// status implements the Status interface
+type status struct {
+	message string
+}
+
+func (s *status) String() string {
+	return s.message
+}
+
+// NewStatus returns the default implementation of the Status interface
+func NewStatus(m string) *status {
+	return &status{message: m}
+}
+
+// StatusStream is an interface for publishing status updates while executing
+// an Empire action.
+type StatusStream interface {
+	// Publish publishes an update to the status stream
+	Publish(Status)
+
+	// Done finalizes the status stream
+	Done(error)
+}
+
+type SubscribableStream interface {
+	Subscribe() <-chan Status
+	Error() error
```

Comment:
Minor, but from what I've seen, the standard convention for adding a method that returns an error like this is to call it `Err()` (see [bufio#Scanner.Err](https://golang.org/pkg/bufio/#Scanner.Err), [sql#Rows.Err](https://golang.org/pkg/database/sql/#Rows.Err), etc).